airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/declarative_source_adapter.py

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+
+from logging import Logger
+from typing import Any, List, Mapping
+
+from airbyte_cdk.models import AirbyteConnectionStatus
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_protocol.models import ConnectorSpecification
+
+
+class DeclarativeSourceAdapter(YamlDeclarativeSource):
+    def __init__(self, source: AbstractSource) -> None:
+        super().__init__(path_to_yaml="manifest.yaml")
+        self._source = source
+        self._set_adapted_methods()
+
+    def spec(self, logger: Logger) -> ConnectorSpecification:
+        return self._source.spec(logger)
+
+    def check(self, logger: Logger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
+        return self._source.check(logger, config)
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        return self._source.streams(config)
+
+    def _validate_source(self) -> None:
+        """Skipping manifest validation as it can be incomplete when use adapter"""
+        return
+
+    def _set_adapted_methods(self) -> None:
```

Comment:
can you add a comment explaining what this is for?

Reply:
Added, thanks for the suggestion!